Signal/src/ViewControllers/DebugUI/DebugUIFileBrowser.swift

```diff
@@ -0,0 +1,380 @@
+//
+//  Copyright (c) 2018 Open Whisper Systems. All rights reserved.
+//
+
+class DebugUIFileBrowser: OWSTableViewController {
+
+    // MARK: Dependencies
+    var fileManager: FileManager {
+        return FileManager.default
+    }
+
+//    let logTag = "[DebugUIFileBrowser]"
+
+    // MARK: Overrides
+    let fileURL: URL
+
+    init(fileURL: URL) {
+        self.fileURL = fileURL
+
+        super.init(nibName: nil, bundle: nil)
+
+        self.contents = buildContents()
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        let titleLabel = UILabel()
+        titleLabel.text = "\(fileURL)"
+        titleLabel.sizeToFit()
+        titleLabel.textColor = UIColor.white
+        titleLabel.lineBreakMode = .byTruncatingHead
+        self.navigationItem.titleView = titleLabel
+    }
+
+    fileprivate func updateContents() {
+        self.contents = buildContents()
+        self.tableView.reloadData()
+    }
+
+    override func viewWillAppear(_ animated: Bool) {
+        // In case files were added / removed in child view controller
+        updateContents()
+        super.viewWillAppear(animated)
```

Comment:
I usually like to do the super first in these UIKit lifecycle methods unless there's a specific reason to do otherwise.  We always want to call super and by placing it first its easier to nice if its missing.

Reply:
Fixed!